network/kademlia.go

```diff
@@ -303,8 +303,10 @@ func (k *Kademlia) SuggestPeer() (suggestedPeer *BzzAddr, saturationDepth int, c
 	var lastPO int       // the last non-empty PO bin in the iteration
 	saturationDepth = -1 // the deepest PO such that all shallower bins have >= k.MinBinSize peers
 	var pastDepth bool   // whether po of iteration >= depth
-	k.conns.EachBin(k.base, Pof, 0, func(po, size int, f func(func(val pot.Val) bool) bool) bool {
+	k.conns.EachBin(k.base, Pof, 0, func(bin *pot.Bin) bool {
```

Comment:
Is it needed to have bin as a pointer. Could it be just a value?

Reply:
Can be a value. You are right.

Reply:
Why is this better as a value?

Reply:
It should be in the function stack instead the heap. Given that this iterator may be frequently used, it could reduce the number allocations.

Reply:
Actually, I was wrong. Compiler is smart not to escape this value to the heap this value. It is just fine.